packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -997,20 +1002,23 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | BetaTea
 
   /* @conditional-compile-remove(close-captions) */
   private subscribeToCaptionEvents(): void {
-    if (this.call && this.call.state === 'Connected' && _isTeamsMeetingCall(this.call)) {
-      this.call?.feature(Features.TeamsCaptions).on('captionsReceived', this.captionsReceived.bind(this));
-      this.call?.feature(Features.TeamsCaptions).on('isCaptionsActiveChanged', this.isCaptionsActiveChanged.bind(this));
-      this.call?.off('stateChanged', this.subscribeToCaptionEvents.bind(this));
+    if (this.call && this.call.state === 'Connected' && this.context.getState().isTeamsCall) {
+      const captionsFeature = this.call?.feature(Features.Captions).captions as TeamsCaptions;
+      captionsFeature.on('CaptionsReceived', this.captionsReceived.bind(this));
+      captionsFeature.on('CaptionsActiveChanged', this.isCaptionsActiveChanged.bind(this));
+      captionsFeature.on('CaptionLanguageChanged', this.spokenLanguageChanged.bind(this));
+      captionsFeature.on('SpokenLanguageChanged', this.captionLanguageChanged.bind(this));
     }
   }
 
   /* @conditional-compile-remove(close-captions) */
   private unsubscribeFromCaptionEvents(): void {
-    if (this.call && this.call.state === 'Connected' && _isTeamsMeetingCall(this.call)) {
-      this._call?.feature(Features.TeamsCaptions).off('captionsReceived', this.captionsReceived.bind(this));
-      this._call
-        ?.feature(Features.TeamsCaptions)
-        .off('isCaptionsActiveChanged', this.isCaptionsActiveChanged.bind(this));
+    if (this.call && this.call.state === 'Connected' && this.context.getState().isTeamsCall) {
```

Comment:
Is it intentionally we only ever unsubscribe from captions when callstate is connected? What if the call ends and then I call unsubscribe?

Reply:
The captions API can only be accessed when the call is connected. So this is intentional.